x-pack/filebeat/docs/inputs/input-httpjson.asciidoc

```diff
@@ -450,14 +320,590 @@ default credentials from the environment will be attempted via ADC. For more inf
+    {
+      "this_is": "also not kept",
+      "entities": {
+        "id1": {
+          "something": "something"
+        }
+      }
+    },
+    {
+      "this_is": "also not kept",
+      "entities": {
+        "id2": {
+          "something": "something 2"
+        }
+      }
+    }
+  ]
+}
+----
+
+Our config will look like
+
+["source","yaml",subs="attributes"]
+----
+filebeat.inputs:
+- type: httpjson
+  config_version: 2
+  interval: 1m
+  request.url: https://example.com
+  response.split:
+    target: body.alerts
+    type: array
+    split:
+      transforms:
+        - set:
+            target: body.new
+            value: will be added to each
+      target: body.entities
+      type: map
+----
+
+This will output:
+
+["source","json",subs="attributes"]
+----
+[
+  {
+    "something": "something",
+    "new": "will be added for each"
+  },
+  {
+    "something": "something 2",
+    "new": "will be added for each"
+  }
+]
+----
+
+[float]
+==== `cursor`
+
+Cursor is a list of key value objects where an arbitrary values are defined. The values are interpreted as value templates and a default template can be set. Cursor state is kept between input restarts and updated once all the events for a request are published.
+
+Can read state from: [`.last_response.*`, `.last_event.*`].
+
+NOTE: Default templates do not have access to any state, only to functions.
+
+["source","yaml",subs="attributes"]
+----
+filebeat.inputs:
+- type: httpjson
+  config_version: 2
+  interval: 1m
+  request.url: https://api.ipify.org/?format=json
+  response.transforms:
+    - set:
+        target: body.last_requested_at
+        value: '{{.cursor.last_requested_at}}'
+        default: "{{now}}"
+  cursor:
+    last_requested_at:
+      value: '{{now}}'
+  processors:
+    - decode_json_fields
+        fields: [message]
+        target: json
+----
+
+[float]
+==== `api_key`
+
+Deprecated, use `request.transforms`.
+
+[float]
+==== `http_client_timeout`
+
+Deprecated, use `request.timeout`.
+
+[float]
+==== `http_headers`
+
+Deprecated, use `request.transforms`.
+
+[float]
+==== `http_method`
+
+Deprecated, use `request.method`.
+
+[float]
+==== `http_request_body`
+
+Deprecated, use `request.body`.
+
+[float]
+==== `json_objects_array`
+
+Deprecated, use `request.split`.
+
+[float]
+==== `split_events_by`
+
+Deprecated, use `request.split`.
+
+[float]
+==== `no_http_body`
+
+Deprecated.
+
+[float]
+==== `pagination.*`
+
+Deprecated, use `response.pagination`.
+
+[float]
+==== `rate_limit.*`
+
+Deprecated, use `request.rate_limit.*`.
+[float]
+==== `retry.*`
+
+Deprecated, use `request.retry.*`.
+
+[float]
+==== `ssl`
+
+Deprecated, use `request.ssl`.
+
+[float]
+==== `url`
+
+Deprecated, use `request.url`.
+
+[float]
+==== `oauth2.*`
+
+Deprecated, use `auth.oauth2.*`.
+
+==== Request life cycle
+
+
+....
++-------+                  +-------------------+ +---------------------+              +---------------------------+   +---------------+ +---------+
```

Comment:
Does the tool that you used to create this output other formats? Ascii isn't ideal in our docs because it requires the user to scroll to the right to see everything. If you use an image instead, it will be properly resized (though maybe not easy to read). It's worth a try, though, to see how it looks in the output.